### DISCUSSION TITLE
what is still needed for a development team to start their development activities for a specific technical component?

### DISCUSSION THREAD

**Discussion by BerkvensNick** [soilwise-he/SoilWise-documentation#19] [Ideas]
To kick-start the developmental phase, what is still required by a certain developmental team to start developing their technical component?

If possible formulate as "user story" (requirement), `As a < type of user >, I need < some goal > so that < some reason >`

e.g.:
- As a developer for the harvesting component, I need a cluster/VM so that I can start running a Docker container with code to harvest
- As a developer for the LLM component, I need "relevant" metadata (CORDIS, Zenodo, INspire, ...)  so that I can experiment extracting geographical entities

**Comment by pvgenuchten** [comment]
A question came up in the last meeting where CI-CD processes will run. The current no-cost program of github has limited CI-CD resources. So preferably CI-CD happens at a dedicated infrastructure. If code/configuration is managed on github, i see 2 options to trigger remote ci-cd

- a dedicated ci-cd environment triggered by github commits
- synchronisation of the github repository to an  on premise git repository, which triggers ci-cd

**Reply by BerkvensNick** [reply]
Thanks Paul!
Do you have an idea of the current limitations?
https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration 
The only thing I find is up to 20 concurrent jobs at a time (or 5 macOS)

**Reply by BerkvensNick** [reply]
discussed in sprint refinement (24/04) and do not directly see an issue with the scope/magnitude of this project.
Current development should be possible

Unless possibly in view of after project maintenance/development

Somakanthan will take up with Thorsten

@pvgenuchten ?

**Reply by pvgenuchten** [reply]
we will discuss this with the upcoming integration design

**Reply by pvgenuchten** [reply]
Wetransform uses a system based on [argo workflows](https://argoproj.github.io/workflows/). Experiments will indicate the impact of moving from GitCI to ArgoWF, expectation is minimal
